Userland/Applications/Mail/MailWidget.cpp

```diff
@@ -0,0 +1,484 @@
+                .section = IMAP::FetchCommand::DataItem::Section {
+                    .type = IMAP::FetchCommand::DataItem::SectionType::HeaderFields,
+                    .headers = { { "Subject" } }
+                }
+            },
+            IMAP::FetchCommand::DataItem {
+                .type = IMAP::FetchCommand::DataItemType::BodySection,
+                .section = IMAP::FetchCommand::DataItem::Section {
+                    .type = IMAP::FetchCommand::DataItem::SectionType::HeaderFields,
+                    .headers = { { "From" } }
+                }
+            }
+        }
+    };
+    // clang-format on
+
+    auto fetch_response = m_imap_client->fetch(fetch_command, false)->await().release_value();
+
+    if (response.status() != IMAP::ResponseStatus::OK) {
+        dbgln("Failed to retrieve subject/from for e-mails. The server says: '{}'", response.response_text());
+        GUI::MessageBox::show_error(window(), String::formatted("Failed to retrieve e-mails. The server says: '{}'", response.response_text()));
+        return;
+    }
+
+    Vector<InboxEntry> active_inbox_entries;
+
+    for (auto& fetch_data : fetch_response.data().fetch_data()) {
+        auto& response_data = fetch_data.get<IMAP::FetchResponseData>();
+        auto& body_data = response_data.body_data();
+
+        auto data_item_has_header = [](IMAP::FetchCommand::DataItem const& data_item, String const& search_header) {
+            return data_item.section.has_value()
+                && data_item.section->type == IMAP::FetchCommand::DataItem::SectionType::HeaderFields
+                && data_item.section->headers.has_value()
+                && data_item.section->headers->find_if([&search_header](String const& header) {
+                       return header == search_header;
+                   })
+                != data_item.section->headers->end();
+        };
+
+        auto subject_iterator = body_data.find_if([&data_item_has_header](Tuple<IMAP::FetchCommand::DataItem, Optional<String>>& data) {
+            const auto data_item = data.get<0>();
+            return data_item_has_header(data_item, "Subject");
+        });
+
+        VERIFY(subject_iterator != body_data.end());
+
+        auto from_iterator = body_data.find_if([&data_item_has_header](Tuple<IMAP::FetchCommand::DataItem, Optional<String>>& data) {
+            const auto data_item = data.get<0>();
+            return data_item_has_header(data_item, "From");
+        });
+
+        VERIFY(from_iterator != body_data.end());
+
+        auto& subject_value = subject_iterator->get<1>().value();
+        auto subject_parts = subject_value.split(':');
+        auto subject = subject_parts.last().trim_whitespace();
+
+        if (subject.is_empty())
+            subject = "(no subject)";
+
+        auto& from_value = from_iterator->get<1>().value();
+        auto from_parts = from_value.split(':');
+        auto from = from_parts.last().trim_whitespace();
```

Comment:
This is insufficient, long lines wrap: https://datatracker.ietf.org/doc/html/rfc2822#section-2.2.3